### PR TITLE
added new websockets page

### PR DESCRIPTION
**Pull request by checklyalex**
## Affected Components
* [ ] Content & Marketing
* [ ] Pricing
* [ ] Test
* [x] Docs
* [ ] Learn
* [ ] Other

## Pre-Requisites
* [ ] Code is linted (`npm run lint`)

### CODE REVIEW

File: site/content/docs/multistep-checks/websockets.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    parent: "Multistep API checks"
+    identifier: "multistep-checks-websockets"
+cli: true
+---
+
+WebSockets provide a way to establish a persistent, full-duplex communication channel that operates over a single, long-held request, enabling real-time data transfer between a client and a server.
+
+To create a WebSocket check in Checkly, you'll need to write a script that establishes a WebSocket connection, sends messages, and validates responses.
+
+## Set Up the WebSocket Connection
```
Comment (ragog) [inline site/content/docs/multistep-checks/websockets.md:15]: @checklyalex this is too drawn out, let's not have a chapter for each small bit of the check. Let's instead have two chapters, one for the generic websocket check code, with comments, an one for the example.

File: site/content/docs/multistep-checks/websockets.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+Don't forget the essentials:
+
+- Error Handling: Implement robust error handling for scenarios like connection failures and timeouts.
+- Message Validation: Add logic to validate incoming messages to ensure the data integrity.
```
Comment (ragog) [inline site/content/docs/multistep-checks/websockets.md:157]: @checklyalex nit: isn't the "the" extra here? "Data integrity"

File: site/content/docs/multistep-checks/websockets.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+- Closing Connections: Always close the WebSocket connection gracefully to avoid resource leaks.
+
+Monitoring WebSocket connections is crucial for real-time applications. With Checkly, you can easily set up checks to ensure your WebSocket connections are reliable and performant. Adjust the example script to fit your specific use case and continuously monitor your WebSocket interactions.
```
Comment (ragog) [inline site/content/docs/multistep-checks/websockets.md:160]: @checklyalex feels extra, let's drop this

File: site/content/docs/multistep-checks/websockets.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  console.log('WebSocket connection closed.');
+});
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+## Example: Monitoring a WebSocket Connection
+Here's the complete example script that demonstrates a WebSocket check. It sends a subscription message, prints out the echo'd response, handles different types of incoming messages and closes the connection after 5 seconds.
+
+{{< tabs "Full WebSocket Script" >}}
```
Comment (ragog) [inline site/content/docs/multistep-checks/websockets.md:92]: @checklyalex question: wouldn't we want some assertions in the check? Or would you say they are superfluous?

**Review by ragog — CHANGES_REQUESTED** [review]
@checklyalex left a couple of comments on things we should address.

File: site/content/docs/multistep-checks/websockets.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+title: WebSockets
+weight: 20
+menu:
+  resources:
+    parent: "Multistep API checks"
+    identifier: "multistep-checks-websockets"
+cli: true
+---
+
+WebSockets provide a way to establish a persistent, full-duplex communication channel that operates over a single, long-held request, enabling real-time data transfer between a client and a server.
```
Comment (tnolet) [inline site/content/docs/multistep-checks/websockets.md:11]: No need to explain what websockets are and how they work. This is not wikipedia

File: site/content/docs/multistep-checks/websockets.md (hunk 3 above)
Comment (tnolet) [inline site/content/docs/multistep-checks/websockets.md:160]: No need to add the sales pitch at the end. These are reference docs.

**Review by tnolet — DISMISSED** [review]
Nice addition, just drop the wikipedia and sales pitch 😄

**Review by ragog — APPROVED** [review]
@checklyalex LGTM. Let's keep an eye out for better examples if we come across them.

**Pull request merged**